### PR TITLE
AP-92: Add abstraction layer for DOM APIs

### PR DESCRIPTION
**Pull request by chrisdevereux**

### CODE REVIEW

File: docs-site/src/pages/docs/interacting-with-the-browser.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        </p>
+      </Section>
+      <Section title="Performing page-level actions">
+        <p>
+          In addition to getting information about page state, we might want to perform actions in a cross-environment way. Bright-js-framework provides a number of action services for this purpose.
+        </p>
+        <p>
+          To programatically navigate:
+        </p>
+        <CodeFile path="src/components/ControlPanel.tsx">
```
Comment (richard-hulm) [inline docs-site/src/pages/docs/interacting-with-the-browser.js]: I think this is the wrong codeblock being referenced?

File: docs-site/src/pages/docs/interacting-with-the-browser.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+          {require('raw!../../../../examples/plugin-with-injection/src/components/ControlPanel.tsx')}
+        </CodeFile>
+        <p>
+          It is important to understand the difference between injected objects and services. Although the APIs are similar, they serve very different purposes. Objects injected by plugins are shared across the whole application. This differs from services, which are unique to each controller. So for an event system, which requires all clients to share a single underlying object, we would use a plugin to provide the event manager. For a form-binding helper, which is a piece of extracted controller logic, we would use a service.
```
Comment (richard-hulm) [inline docs-site/src/pages/docs/interacting-with-the-browser.js]: Isn't this all copy and pasted from extending-with-plugins.js

File: docs-site/src/pages/docs/interacting-with-the-browser.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        </p>
+        <p>
+          To abstract the details of the key away from the users of our plugin, we've exported a decorator named
+          <code>@eventManager</code>. In this case, this is simply <code>@inject</code> with a prefilled key. This
+          does, however, give us (the plugin authors) the freedom to change how the clients of the plugin receive
+          the EventManager. In the future, we may wish to do more than just a simple inject; we may also wish to
+          utilise <code>@service</code> capabilities, for example.
+        </p>
+      </Section>
+      <Section title="Navigating to new pages">
```
Comment (richard-hulm) [inline docs-site/src/pages/docs/interacting-with-the-browser.js]: Again, this section seems to be copied and pasted?

**Review by richard-hulm — COMMENTED** [review]
Looking back through other projects, we make a lot of calls to window to close the current tab, or get the size of the window or even use confirmation boxes, among other things.

How are we handling calls to things such as this? Are we calling them directly? I assume not as this wouldn't work brilliantly with SSR

**Comment by chrisdevereux** [comment]
There are a lot of DOM APIs we could and (probably will want to) include in this, but doing them all properly now would take a long time. Was thinking we could add other calls as part of user-facing stories, though. Mostly, the idea here is to get the infrastructure for it in place.

File: docs-site/src/pages/docs/interacting-with-the-browser.js (hunk 2 above)
Comment (chrisdevereux) [reply docs-site/src/pages/docs/interacting-with-the-browser.js]: 🤔

**Review by richard-hulm — APPROVED** [review]

**Pull request merged**